File: coding/kotlin/code/builder.java
class Builder {
    int a;
    int b;
    String c;

    Builder setA(int a) {
        this.a = a;
        return this;
    }

    Builder setB(int b) {
        this.b = b;
        return this;
    }

    Builder setC(String c) {
        this.c = c;
        return this;
    }
}

static Builder buildExample() {
    Builder builder = new Builder();
    int i = 0;
    builder.setA(i);
    i+=5;
    return builder.setB(i).setC("Done");
}

File: coding/kotlin/code/builder.kt
class Builder(var a: Int, var b: Int, var c: String)

fun build(builder: Builder.() -> Unit): Builder
    = with(Builder()) { builder() }

fun exampleBuild() = build {
    val i = 0
    a = i
    i += 5
    b = i
    c = "Done"
}